velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -209,4 +209,25 @@ struct LastDayFunction {
   }
 };
 
+template <typename T>
+struct DateSubFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Date>& result,
+      const arg_type<Date>& date,
+      const int32_t value) {
+    constexpr int32_t kMin = std::numeric_limits<int32_t>::min();
+    int32_t subValue = 0;
```

Comment:
This variable is not used outside of if-else block. Please, move it inside if and else bodies or remove altogether.